src/text.c

```diff
@@ -1438,3 +1760,412 @@ u8 RenderTextFont9(u8 *pixels, u8 fontId, u8 *str)
+}
+
+u8 GetKeypadIconTileOffset(u8 keypadIconId)
+{
+    return gKeypadIcons[keypadIconId].tile_offset;
+}
+
+u8 GetKeypadIconWidth(u8 keypadIconId)
+{
+    return gKeypadIcons[keypadIconId].width;
+}
+
+u8 GetKeypadIconHeight(u8 keypadIconId)
+{
+    return gKeypadIcons[keypadIconId].height;
+}
+
+void DecompressGlyphFont0(u16 glyphId, bool32 isJapanese)
+{
+    const u16* glyphs;
+
+    if (isJapanese == 1)
+    {
+        glyphs = gFont0JapaneseGlyphs + (0x100 * (glyphId >> 0x4)) + (0x8 * (glyphId & 0xF));
+        DecompressGlyphTile(glyphs, (u16 *)gGlyphInfo);
+        DecompressGlyphTile(glyphs + 0x80, (u16 *)(gGlyphInfo + 0x40));
+        gGlyphInfo[0x80] = 8;     // gGlyphWidth
+        gGlyphInfo[0x81] = 12;    // gGlyphHeight
+    }
+    else
+    {
+        glyphs = gFont0LatinGlyphs + (0x10 * glyphId);
+        DecompressGlyphTile(glyphs, (u16 *)gGlyphInfo);
+        DecompressGlyphTile(glyphs + 0x8, (u16 *)(gGlyphInfo + 0x40));
+        gGlyphInfo[0x80] = gFont0LatinGlyphWidths[glyphId];
+        gGlyphInfo[0x81] = 13;
+    }
+}
+
+u32 GetGlyphWidthFont0(u16 glyphId, bool32 isJapanese)
+{
+    if (isJapanese == TRUE)
+        return 8;
+    else
+        return gFont0LatinGlyphWidths[glyphId];
+}
+
+void DecompressGlyphFont1(u16 glyphId, bool32 isJapanese)
+{
+    const u16* glyphs;
+
+    if (isJapanese == TRUE)
+    {
+        int eff;
+        glyphs = gFont1JapaneseGlyphs + (0x100 * (glyphId >> 0x4)) + (0x8 * (glyphId & (eff = 0xF)));  // shh, no questions, only matching now
+        DecompressGlyphTile(glyphs, (u16 *)gGlyphInfo);
+        DecompressGlyphTile(glyphs + 0x80, (u16 *)(gGlyphInfo + 0x40));    // gUnknown_03002FD0
+        gGlyphInfo[0x80] = 8;     // gGlyphWidth
+        gGlyphInfo[0x81] = 16;    // gGlyphHeight
+    }
+    else
+    {
+        glyphs = gFont1LatinGlyphs + (0x20 * glyphId);
+        DecompressGlyphTile(glyphs, (u16 *)gGlyphInfo);
+        DecompressGlyphTile(glyphs + 0x8, (u16 *)(gGlyphInfo + 0x20));
+        DecompressGlyphTile(glyphs + 0x10, (u16 *)(gGlyphInfo + 0x40));
+        DecompressGlyphTile(glyphs + 0x18, (u16 *)(gGlyphInfo + 0x60));
+        gGlyphInfo[0x80] = gFont1LatinGlyphWidths[glyphId];
+        gGlyphInfo[0x81] = 14;
+    }
+}
+
+u32 GetGlyphWidthFont1(u16 glyphId, bool32 isJapanese)
+{
+    if (isJapanese == TRUE)
+        return 8;
+    else
+        return gFont1LatinGlyphWidths[glyphId];
+}
+
+void DecompressGlyphFont2(u16 glyphId, bool32 isJapanese)
+{
+    const u16* glyphs;
+    int i;
+    u8 lastColor;
+
+    if(isJapanese == TRUE)
+    {
+        if(glyphId == 0)
+        {
+            lastColor = GetLastTextColor(2);
+
+            for(i = 0; i < 0x80; i++)
+            {
+                gGlyphInfo[i] = lastColor | lastColor << 4;
+                // Game Freak, please. writing the same values over and over...
+                gGlyphInfo[0x80] = 10;
+                gGlyphInfo[0x81] = 12;
+            }
+        }
+        else // _080065F8
```

Comment:
remove leftover debugging comments.  There are some address comments, along with a bunch of gUnknown comments after DecompressGlyphTile() calls.